src/app/clusters/commodity-tariff-server/commodity-tariff-server.h

```diff
@@ -0,0 +1,172 @@
+                                                                           DataModel::List<const uint32_t> & dayEntryIDs,
+                                                                           Structs::TariffComponentStruct::Type & aTariffComponent) = 0;
+
+    // ------------------------------------------------------------------
+#if 0
+    // Attribute  getters
+    virtual DataModel::Nullable<Structs::TariffInformationStruct::Type> & GetTariffInfo()        = 0;
+    virtual DataModel::Nullable<Globals::TariffUnitEnum> & GetTariffUnit()                       = 0;
+    virtual DataModel::Nullable<uint32_t> & GetStartDate()                                       = 0;
+    virtual DataModel::Nullable<int16_t> & GetDefaultRandomizationOffset()                       = 0;
+    virtual DataModel::Nullable<DayEntryRandomizationTypeEnum> & GetDefaultRandomizationType()   = 0;
+    virtual DataModel::List<Structs::CalendarPeriodStruct::Type> & GetCalendarPeriods()          = 0;
+    virtual DataModel::List<Structs::DayPatternStruct::Type> & GetDayPatterns()                  = 0;
+    virtual DataModel::List<Structs::DayStruct::Type> & GetIndividualDays()                      = 0;
+    virtual DataModel::List<Structs::DayEntryStruct::Type> & GetDayEntries()                     = 0;
+    virtual DataModel::List<Structs::TariffPeriodStruct::Type> & GetTariffPeriods()              = 0;
+    virtual DataModel::List<Structs::TariffComponentStruct::Type> & GetTariffComponents()        = 0;
+    virtual DataModel::Nullable<Structs::DayStruct::Type> & GetCurrentDay()                      = 0;
+    virtual DataModel::Nullable<Structs::DayStruct::Type> & GetNextDay()                         = 0;
+    virtual DataModel::Nullable<Structs::DayEntryStruct::Type> & GetCurrentDayEntry()            = 0;
+    virtual DataModel::Nullable<Structs::DayEntryStruct::Type> & GetNextDayEntry()               = 0;
+    virtual DataModel::Nullable<uint32_t> & GetCurrentDayEntryDate()                             = 0;
+    virtual DataModel::Nullable<uint32_t> & GetNextDayEntryDate()                                = 0;
+    virtual DataModel::List<Structs::TariffComponentStruct::Type> & GetCurrentTariffComponents() = 0;
+    virtual DataModel::List<Structs::TariffComponentStruct::Type> & GetNextTariffComponents()    = 0;
+#else
+    #define X(attrName, attrType) \
+        virtual attrType& Get##attrName() = 0;
+    COMMODITY_TARIFF_PRIMARY_ATTRIBUTES
+    COMMODITY_TARIFF_CURRENT_ATTRIBUTES
+    #undef X
+#endif
+
+private:
+    // Generate setters
+    #define X(attrName, attrType) \
+        virtual CHIP_ERROR Set##attrName(const attrType& newValue) { \
+            return CHIP_NO_ERROR; \
+        }
+    COMMODITY_TARIFF_CURRENT_ATTRIBUTES
+    #undef X
+
+    // Helper function for change reporting
+    virtual void ReportAttributeChange(uint16_t attributeId) = 0;
```

Comment:
Attribute IDs are generally 32-bit in Matter....  What does this actually report and to whom?